src/layout.rs

```diff
@@ -215,11 +218,18 @@ struct WaitingState<T: 'static> {
     tap: &'static Action<T>,
     config: HoldTapConfig,
 }
-enum WaitingAction {
+
+/// Actions that can be triggered for a key configured for HoldTap.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum WaitingAction {
+    /// Trigger the holding event.
     Hold,
+    /// Trigger the tapping event.
     Tap,
+    /// Do not trigger any event.
     NoOp,
```

Comment:
Do you want to overload NoOp or keep the Option behavior? It seems to like above you preferred a "fallback" meaning to the public API and a "do nothing this tick" to the private usage. I'm ok with this. It could have a doc-comment for the public usage and a non-doc comment for the private behavior as well.

Reply:
I don't really know what is better. My main concern is easy to use, for the library user, API. It is better to have consistent code for maintenance, having a different behavior internally and externally is dangerous.

I badly chosen `NoOp` for this case (it was a private API, so I didn't think too much on the naming), as, internally, it means "don't choose now".

I think there is several posibilities. Some of them:
* change the name `NoOp` to something like `ChooseLater` (I can't find a proper name), and the function don't return an option.
* remove `NoOp`, and None mean "chose later"
* keep `NoOp`, but change the handling to not generate any event in this case, and removing the Waiting thing. This solution seems to be want you wanted to do. The internal code must be changed to handle this case.
* maybe something else.

Reply:
In https://github.com/TeXitoi/keyberon/pull/82#discussion_r851668067 I think you "want" the function to say "don't generate any event, just forget about it" by returning NoOp. So I was talking of this feature.

Reply:
Ok, I just pushed a change that includes the "don't generate an event" behavior along with an additional test in layout to ensure it works as expected.